clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/jobs/local/JobExecutorLocal.groovy

```diff
@@ -0,0 +1,172 @@
+          CommandLine commandLine
+
+          if (jobRequest.tokenizedCommand) {
+            log.info("Executing $jobId with tokenized command: $jobRequest.tokenizedCommand")
+
+            // Grab the first element as the command.
+            commandLine = new CommandLine(jobRequest.tokenizedCommand[0])
+
+            // Treat the rest as arguments.
+            String[] arguments = Arrays.copyOfRange(jobRequest.tokenizedCommand.toArray(), 1, jobRequest.tokenizedCommand.size())
+
+            commandLine.addArguments(arguments, false)
+          } else {
+            log.info("No tokenizedCommand specified for $jobId.")
+
+            throw new IllegalArgumentException("No tokenizedCommand specified for $jobId.")
+          }
+
+          DefaultExecuteResultHandler resultHandler = new DefaultExecuteResultHandler()
+          ExecuteWatchdog watchdog = new ExecuteWatchdog(timeoutMinutes * 60 * 1000){
+            @Override
+            void timeoutOccured(Watchdog w) {
+              // If a watchdog is passed in, this was an actual time-out. Otherwise, it is likely
+              // the result of calling watchdog.destroyProcess().
+              if (w) {
+                log.info("Job $jobId timed-out (after $timeoutMinutes minutes).")
+
+                cancelJob(jobId)
+              }
+
+              super.timeoutOccured(w)
+            }
+          }
+          Executor executor = new DefaultExecutor()
+          executor.setStreamHandler(pumpStreamHandler)
+          executor.setWatchdog(watchdog)
+          executor.execute(commandLine, resultHandler)
+
+          // Give the job some time to spin up.
+          sleep(500)
+
+          jobIdToHandlerMap.put(jobId, [
+            handler: resultHandler,
+            watchdog: watchdog,
+            stdOutAndErr: stdOutAndErr
+          ])
+        }
+      }
+    )
+
+    return jobId
+  }
+
+  @Override
+  JobStatus updateJob(String jobId) {
+    try {
+      log.info("Polling state for $jobId...")
+
+      if (jobIdToHandlerMap[jobId]) {
+        JobStatus jobStatus = new JobStatus(id: jobId)
+
+        DefaultExecuteResultHandler resultHandler
+        ByteArrayOutputStream stdOutAndErr
+
+        jobIdToHandlerMap[jobId].with {
+          resultHandler = it.handler
+          stdOutAndErr = it.stdOutAndErr
+        }
+
+        String logsContent = new String(stdOutAndErr.toByteArray())
+
+        if (resultHandler.hasResult()) {
+          log.info("State for $jobId changed with exit code $resultHandler.exitValue.")
+
+          if (!logsContent) {
+            logsContent = resultHandler.exception ? resultHandler.exception.message : "No output from command."
+          }
+
+          if (resultHandler.exitValue == 0) {
+            jobStatus.state = JobStatus.State.COMPLETED
+            jobStatus.result = JobStatus.Result.SUCCESS
+          } else {
+            jobStatus.state = JobStatus.State.CANCELED
```

Comment:
Why `CANCELED`? I think `130` is the de-facto `ctrl-c` exit code, but a failure doesn't usually imply a cancellation.